### DISCUSSION TITLE
Getting Unpublished Version of Object

### DISCUSSION THREAD

**Discussion by cancan101** [pimcore/pimcore#17056] [Q&A / Support]
How do I get the current version of an Object, including any unpublished changes?
```php
$myObject = \Pimcore\Model\DataObject\MyObject::getById($id, true);
```
returns the DB version.

/CC @markus-moser (related to https://github.com/orgs/pimcore/discussions/16962)

**Comment by cancan101** [comment]
I believe this works, though not sure if there are any gotchas I should be aware of:
```php
$user = \Pimcore\Tool\Admin::getCurrentUser();
$latestVersion = $myObject->getLatestVersion($user?->getId());
if ($latestVersion) {
    $latestObj = $latestVersion->loadData();
    if ($latestObj instanceof \Pimcore\Model\DataObject\MyObject) {
        $myObject = $latestObj;
    }
}
```
took inspiration from: https://github.com/pimcore/admin-ui-classic-bundle/blob/d191bcd0184f77e3ac9d25270ee02a4fcc2496aa/src/Controller/Admin/DataObject/DataObjectController.php#L2025-L2038

**Reply by markus-moser** [reply]
Yes with this you get the latest version of the object! But be aware that this could be also the auto saved draft version of the logged in user (see https://github.com/pimcore/pimcore/blob/11.x/models/Element/Traits/VersionDaoTrait.php#L35)

If you would like to skip auto saved drafts you would need to create your own version of `getLatestVersion` with a custom query.

**Reply by cancan101** [reply]
Do I also need to pass `includingPublished=true` to include the published version if there are no drafts? 

How does autosaved work when passing in the `userId`? Are auto saved scoped by user? I actually am fine including autosaved drafts as long as for the current user.

**Reply by markus-moser** [reply]
Yes the auto saved drafts are scoped by user and yes if you want to include the published version you would need to pass `includingPublished=true`.
